core/libdeno/deno.h

```diff
@@ -84,7 +84,7 @@ void deno_execute(Deno* d, void* user_data, const char* js_filename,
 // releasing its memory.)
 //
 // If a JS exception was encountered, deno_last_exception() will be non-NULL.
-void deno_respond(Deno* d, void* user_data, deno_buf buf);
+void deno_respond(Deno* d, void* user_data, deno_buf buf, int* promise_id);
```

Comment:
Where does promise_id come from? Update docs in deno.h

Reply:
It comes from pre_dispatch. When the dispatch returns a Op::Async we create a promise id in rust and immediately return it via response/deno_respond. This means that a send/dispatch call will return a `Uint8Array | null | number` value. With numbers always being promise ids. I'll make sure to update docs to reflect this change.